pagefaultgames/pokerogue
# [P1] Fix wrong local save being deleted when creating a new run

## What are the changes the user will see?
No more accidentally erased save when creating a new run offline
One less trigger for save rollbacks when playing online


## Why am I making these changes?
Losing runs or progress for no good reason is not great
The problem happens when creating a new run in a slot that has already a save file.
The game tries to delete the local save file in that slot before making the new one, but instead it deletes the save in the slot that was last played after a save & quit, otherwise it defaults to the first slot
Offline the run is fully lost, online there could be progress rollback if the run wasn't synced up with the server.

Note that this bug is part of why online players get rollbacks on their saves, but it's not the only reason. For example save & quitting or refreshing the page just after creating a new run seem to always result in all local saves getting erased, causing rollbacks for non synced runs.
This PR does not address those other issues so rollbacks will still happen

 



## What are the changes from a developer perspective?
`deleteSession` and `tryClearSession` now use the provided `slotId` parameter instead of `scene.sessionSlotId`

Note that for `tryClearSession` it does not change anything since it's only called after a run is finished (either lost or won), so the slotId and `scene.sessionSlotId` are the same in this case.
Which is also why my examples are all about creating new games




### Screenshots/Videos
Before: offline - slot 1 getting deleted when creating a run in slot 3

https://github.com/user-attachments/assets/fd06fef9-fa70-40bf-9d98-c645bab040aa

After: offline - no deleted save after creating a new run

https://github.com/user-attachments/assets/43af4871-d229-44d5-9e54-96159898c790


Before: online - slot 1 getting rolled back when creating a run in slot 3

https://github.com/user-attachments/assets/9e6b9af0-5d57-4b0b-b1a4-b2935683e02f

After: online - no rollback for slot 1 after creating a new run in slot 2

https://github.com/user-attachments/assets/1420e213-efe0-4bae-9a52-91a9daeac970






## How to test the changes?
* Oflline:
  * Make sure you have a run in the first slot and at least another slot
  * New game
  * Save in a non empty slot, but not the first one
  * Save & quit
  * All saves should still be here

Online:
  * Make sure you have a run in the first slot
  * Play a couple waves and refresh (not save & quit) so that the save is not synced with the server
  * New Game
  * Save in a non empty slot, but not the first one
  * Play at least one wave then refresh (save & quitting or refreshing before wave 2 would cause an unrelated rollback)
  * All saves should still be here, with no rollback on any slots






## Checklist
- [ ] There is no overlap with another PR? Partial overlap / very limited conflict with #4583
